### PR TITLE
[chore](backup) limit the involved tablets in a backup job

### PR DESCRIPTION
**Pull request by w41ter**
and to avoid FE OOM caused by saving too much metadata.

Assuming the average tablet size is 50MB, the default value of 300000 can support 14TB of data per backup job.

The docs PR: https://github.com/apache/doris-website/pull/1056

### CODE REVIEW

**Comment by w41ter** [comment]
run buildall

**Review by gavinchou — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/backup/BackupJob.java [hunk 1]
```diff
@@ -523,6 +523,16 @@ private void prepareAndSendSnapshotTask() {
             }
         }
 
+        // Limit the max num of tablets involved in a backup job, to avoid OOM.
+        if (unfinishedTaskIds.size() > Config.max_backup_tablets_per_job) {
+            String msg = String.format("the num involved tablets %d exceeds the limit %d"
+                    + "change config `max_backup_tablets_per_job` to change this limitation",
+                    unfinishedTaskIds.size(), Config.max_backup_tablets_per_job);
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/backup/BackupJob.java:530]: Please provide users a fe oom hint.

**Comment by w41ter** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by gavinchou — APPROVED** [review]

**Pull request merged**